### PR TITLE
[MLU] add where_grad and where(fp16)

### PR DESCRIPTION
**Pull request by claire199705**
add where_grad and where(fp16)

### CODE REVIEW

File: backends/mlu/kernels/where_kernel.cc [hunk 1]
```diff
@@ -38,7 +39,76 @@ void WhereKernel(const Context& dev_ctx,
...
 }  // namespace custom_kernel
+}  // namespace custom_kernel
+
+PD_REGISTER_PLUGIN_KERNEL(where,
+                          mlu,
+                          ALL_LAYOUT,
+                          custom_kernel::WhereKernel,
+                          int32_t,
+                          float,
+                          phi::dtype::float16) {}
```
Comment (ShawnNew) [inline backends/mlu/kernels/where_kernel.cc:106]: 前向fp16和反向的测例覆盖一下

File: backends/mlu/kernels/where_kernel.cc [hunk 2]
```diff
@@ -38,7 +39,76 @@ void WhereKernel(const Context& dev_ctx,
...
 
-PD_REGISTER_PLUGIN_KERNEL(
-    where, mlu, ALL_LAYOUT, custom_kernel::WhereKernel, int32_t, float) {}
+PD_REGISTER_PLUGIN_KERNEL(where_grad,
```
Comment (cifar10) [inline backends/mlu/kernels/where_kernel.cc:108]: （1）单元测例中需要开启check_grad项，注意self.__class__.no_need_check_grad 。
（2）单元测例的反向运行在了cpu上。

**Comment by claire199705** [comment]
124: I0711 19:45:28.493150 96965 init.cc:239] ENV [CUSTOM_DEVICE_ROOT]=/pp-ws/PaddleCustomDevice/backends/mlu/build/python/paddle_custom_device/
124: I0711 19:45:28.493181 96965 init.cc:145] Try loading custom device libs from: [/pp-ws/PaddleCustomDevice/backends/mlu/build/python/paddle_custom_device/]
124: I0711 19:45:28.539585 96965 custom_device.cc:1112] Successed in loading custom runtime in lib: /pp-ws/PaddleCustomDevice/backends/mlu/build/python/paddle_custom_device//libpaddle-custom-mlu.so
124: I0711 19:45:28.541513 96965 custom_kernel.cc:76] Successed in loading 231 custom kernel(s) from loaded lib(s), will be used like native ones.
124: I0711 19:45:28.541579 96965 init.cc:157] Finished in LoadCustomDevice with libs_path: [/pp-ws/PaddleCustomDevice/backends/mlu/build/python/paddle_custom_device/]
124: I0711 19:45:28.541594 96965 init.cc:245] CustomDevice: mlu, visible devices count: 1
124: I0711 19:45:28.793005 96965 program_interpreter.cc:171] New Executor is Running.
124: I0711 19:45:30.409076 96965 interpreter_util.cc:529] Standalone Executor is Used.
124: /opt/py37env/lib/python3.7/site-packages/paddle/fluid/executor.py:1523: UserWarning: There are no operators in the program to be executed. If you pass Program manually, please use fluid.program_guard to ensure the current Program is being used.
124:   warnings.warn(error_info)
1/1 Test #124: test_where_op_mlu ................   Passed    3.27 sec

The following tests passed:
        test_where_op_mlu

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   3.27 sec

**Comment by cifar10** [comment]
LGTM

**Review by cifar10 — COMMENTED** [review]
LGTM

**Review by cifar10 — APPROVED** [review]

**Pull request merged**